rust/lance/src/arrow/json.rs

```diff
@@ -172,6 +172,9 @@ pub struct JsonField {
     #[serde(rename = "type")]
     type_: JsonDataType,
     nullable: bool,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
Not super important but would it be possible to do `skip_serializing_if = "HashMap::is_empty"` and then skip the option?  I'm not sure if that is more or less clear.]

Hmm...maybe this won't work on deserialization.

Reply:
I was thinking we'd keep this as `Option::is_none` as it maintains consistency with how we're currently handling this for schema metadata